src/azure-cli/azure/cli/command_modules/eventgrid/tests/latest/test_eventgrid_commands.py

```diff
@@ -670,6 +670,44 @@ def test_event_subscription_with_storagequeuemessage_ttl(self, resource_group):
 
         self.cmd('az eventgrid event-subscription delete --source-resource-id {source_resource_id} --name {event_subscription_name}')
 
+    @ResourceGroupPreparer(name_prefix='clieventgrid', location='centraluseuap')
+    @StorageAccountPreparer(name_prefix='clieventgrid', location='centraluseuap')
+    def test_event_subscription_with_delivery_identity(self, resource_group):
+        scope = self.cmd('az group show -n {} -o json'.format(resource_group)).get_output_in_json()['id']
+        event_subscription_name = self.create_random_name(prefix='cli', length=40)
+        eventhub_id = '/subscriptions/5b4b650e-28b9-4790-b3ab-ddbd88d727c4/resourceGroups/DevExpRg/providers/Microsoft.eventhub/namespaces/devexpeh/eventhubs/eventhub1'
+
+        self.kwargs.update({
+            'event_subscription_name': event_subscription_name,
+            'eventhub_id': eventhub_id,
+            'location': 'centraluseuap',
+            'scope': scope,
+        })
+
+        self.kwargs['source_resource_id'] = self.cmd('storage account create -g {rg} -n {sa} --sku Standard_LRS -l {location}').get_output_in_json()['id']
+        self.cmd('az storage account update -g {rg} -n {sa} --set kind=StorageV2')
```

Comment:
```suggestion
        self.kwargs['source_resource_id'] = self.cmd('storage account create -g {rg} -n {sa} --sku Standard_LRS --kind StorageV2 -l {location}').get_output_in_json()['id']
```

Reply:
i think the issue is we can't create with the right kind hence done in 2 steps..